exercises/circular-buffer/example.rs

```diff
@@ -4,36 +4,36 @@ pub enum Error {
     FullBuffer,
 }
 
-pub struct CircularBuffer {
-    buffer: Vec<char>,
+pub struct CircularBuffer<T: Default + Clone> {
+    buffer: Vec<T>,
     size: usize,
     start: usize,
     end: usize,
 }
 
-impl CircularBuffer {
+impl<T: Default + Clone> CircularBuffer<T> {
     // this circular buffer keeps an unallocated slot between the start and the end
     // when the buffer is full. 
-    pub fn new(size: usize) -> CircularBuffer {
+    pub fn new(size: usize) -> CircularBuffer<T> {
         CircularBuffer { 
-            buffer: vec!['0'; size + 1], 
+            buffer: vec![T::default(); size + 1], 
```

Comment:
this is true, but this only affects the example solution - students are still free to use solutions that do not have the `Default` constraint.

Can even avoid the `Clone` constraint by not using the `vec![e; n]` macro.

Yeah, since these disadvantages (which are definitely true) are only specific to the example I consider them fine.